pintc/tests/storage/bad_storage_types.pnt

```diff
@@ -2,38 +2,26 @@ union MyUnion = A | B;
 type MyAlias = MyUnion;
 
 storage {
-    a: ({ int, int } => int),
-    b: (int[5] => int),
-    c: (int[]  => int),
-    d: {int, int}[],
-    e: (int => int)[],
-    f: MyAlias,
-    g: MyUnion,
+    a: {int, int}[],
+    b: (int => int)[],
+    c: MyAlias,
+    d: MyUnion,
 }
 
 // parsed <<<
 // union ::MyUnion = A | B;
 // type ::MyAlias = ::MyUnion;
 // storage {
-//     a: ( {int, int} => int ),
-//     b: ( int[5] => int ),
-//     c: ( int[] => int ),
-//     d: {int, int}[],
-//     e: ( int => int )[],
-//     f: ::MyAlias,
-//     g: ::MyUnion,
+//     a: {int, int}[],
+//     b: ( int => int )[],
+//     c: ::MyAlias,
+//     d: ::MyUnion,
 // }
 // >>>
 
 // typecheck_failure <<<
 // type not allowed in storage
-// @65..86: found type ( {int, int} => int ) in storage
+// @65..77: found type {int, int}[] in storage
 // type not allowed in storage
-// @95..110: found type ( int[5] => int ) in storage
-// type not allowed in storage
-// @119..134: found type ( int[] => int ) in storage
-// type not allowed in storage
-// @143..155: found type {int, int}[] in storage
-// type not allowed in storage
-// @164..178: found type ( int => int )[] in storage
```

Comment:
Do we allow vectors in the map's `ty_from` field? I didn't see anything in the Pint Book that says otherwise.

The check below still requires that the vector be of type `bool`, `int`, or `b256`.

```rust
match self {
            Type::Map { ty_from, ty_to, .. } => {
                // We only support maps from these types for now
                (ty_from.is_bool() || ty_from.is_int() || ty_from.is_b256())
                (ty_from.is_bool()
                    || ty_from.is_int()
                    || ty_from.is_b256()
                    || ty_from.is_allowed_in_storage())
                    && ty_to.is_allowed_in_storage()
            }
            Type::Vector { ty, .. } => {
                // We only support vectors of these types for now
                ty.is_bool() || ty.is_int() || ty.is_b256()
            }
            Type::Array { ty, .. } => ty.is_allowed_in_storage(),
            Type::Tuple { fields, .. } => fields
                .iter()
                .fold(true, |acc, (_, field)| acc && field.is_allowed_in_storage()),
            Type::Alias { ty, .. } => ty.is_allowed_in_storage(),
            Type::Custom { .. } => {
                // Disallow custom types since they're ambiguous. Hopefully, by the time we need
                // this method, all custom types are gone.
                false
            }
            _ => true,
        }
...
```

Reply:
don't worry about vectors. They're not officially supported as they still need a lot of work.

Reply:
Alright. So I should remove support and check directly for the approved tuple types?

Reply:
Once you address my other comments about `get_storage_only_ty`, things should go back to normal.